app/controllers/ontology_search_controller.rb

```diff
@@ -39,4 +44,27 @@ def refine_search_response
     end
     @search_response
   end
-end
+
+  def search_in_repository
+    if in_repository?
+      @anded << "repository_id: #{params[:repository_id]}"
+    end
+  end
+
+  def search_with_params
+    if params[:query].present? && in_repository?
+      search_params = params[:query].split(' ')
+      @anded << search_params.delete(search_params.first)
+      @anded = @anded.join(' AND ')
+      if search_params.present?
+        @ored = search_params.join(' OR ')
+        @search_query = @anded + ' AND (' + @ored + ')'
+      end
+      @search_query = @anded
+    elsif params[:query].present? && !in_repository?
+      @search_query = params[:query].split(' ').join(' OR ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.